JesusFilm/core
# refactor: eng 908 searchbar provider refactor

# Description

### Issue

The search bar provider needs to be told how to update it's state like which languages to remove or which to add. This means components that don't care what continent a language is part of still need to calculate this and tell the provider. 
[Link to Linear Todo](https://linear.app/jesus-film-project/issue/ENG-908/searchbar-provider-refactor)

### Solution

The provider should be responsible for updating it's own state. I.e. let components that want to refine by a language, only worry about refining by a language. In the case where a component wants to add a language to a specific continent, it can tell the context this - otherwise the context will update it's state accordingly

### Additional information
The expensive api call for all continent languages is triggered on a user interaction and so on complete, is given to the context. The context needs this to maintain it's state. There is a chance that while waiting for this on initial call, a user may do something to apply a refinement that cannot be applied until this call is resolved.

### How to test
Regression test all interactions that add and remove language refinements to ensure they still behave as before
* select a language from dropdown
* select another language from the dropdown
* select first suggestion to apply one language
* select second suggestion to apply two languages
* use the language buttons to remove a language
* use the clear all button to remove all languages

In all of the above cases, when a language refinement is applied
1. it should be ticked
2. other copies of that language should be disabled
3. the refinement name should appear in the url
